src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplate.java

```diff
@@ -412,35 +400,40 @@ public void onNext(PullResponseItem item) {
 
     }
 
-    @Restricted(NoExternalUse.class)
-    public DockerTransientNode provisionNode(DockerAPI api, TaskListener listener) throws IOException, Descriptor.FormException, InterruptedException {
+    public class ContainerCommandCreator implements Serializable{
```

Comment:
Hi @ndeloof, I had to add this ```ContainerCommandCreator```, since the CreateContainerCmd is not Serializable and therefore when passing it to the anonymous ```DelegatingComputerLauncher(new JNLPLauncher()))``` through the ```createLauncher``` method, Jenkins will fail serialization. 
So the trick was to do it lazily using the ```DockerTemplate``` which is serializable. So this class is a holder for the uniqueContainerName and when needed, the it creates the CreateContainerCmd and fills it will all the required data from the DockerTemplate instance.